### PR TITLE
[Issue 6283][Offloaders] Offload policies per ns with deletionLag & threshold

### LINKED ISSUE #6283: [offloaders] Define treshold and deletionMaxLag per ns or use the default one.
**Is your feature request related to a problem? Please describe.**
The OffloadPolices per ns is just merged. But it's missing a point on values to trigger auto offloading: we want to define tresholdMaxBytes and deletionMaxLag per ns or use the default one configured.

**Describe the solution you'd like**
Add  tresholdMaxBytes and deletionMaxLag in OffloadPolices and use them to defines triggers.

### PR DESCRIPTION
**Pull request by KannarFr**
Fixes #6283 

### Modifications

Define and use custom deletionLag and threshold for offloadpolicies per ns

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is already covered by existing tests, such as .

### Does this pull request potentially affect one of the following parts:

  - The public API: yes, new fields to add to offloadpolicies
  - The admin cli options: yes, new fields to add to offloadpolicies

### Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? not documented

### CODE REVIEW

**Comment by KannarFr** [comment]
@gaoran10 Hi, can you review this?

**Comment by codelipenghui** [comment]
@gaoran10 Please also help review this PR since you have done lots of work in the namespace level offload policy.

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerConfig.java [hunk 1]
```diff
@@ -417,7 +415,7 @@ public long getRetentionSizeInMB() {
      * @param unit timeunit for lagTime
      */
     public ManagedLedgerConfig setOffloadLedgerDeletionLag(long lagTime, TimeUnit unit) {
```
Comment (codelipenghui) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerConfig.java]: Looks no longer used, can it be deleted?

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerConfig.java [hunk 2]
```diff
@@ -439,7 +437,7 @@ public long getOffloadLedgerDeletionLagMillis() {
      * @param threshold Threshold in bytes at which offload is automatically triggered
      */
     public ManagedLedgerConfig setOffloadAutoTriggerSizeThresholdBytes(long threshold) {
```
Comment (codelipenghui) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerConfig.java]: Looks no longer used, can it be deleted?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 3]
```diff
@@ -1006,19 +1006,6 @@ public void openLedgerFailed(ManagedLedgerException exception, Object ctx) {
...
-                        lag = p.offload_deletion_lag_ms;
-                    }
-                    long bytes = serviceConfig.getManagedLedgerOffloadAutoTriggerSizeThresholdBytes();
-                    if (p.offload_threshold != -1L) {
-                        bytes = p.offload_threshold;
-                    }
-                    managedLedgerConfig.setOffloadLedgerDeletionLag(lag, TimeUnit.MILLISECONDS);
-                    managedLedgerConfig.setOffloadAutoTriggerSizeThresholdBytes(bytes);
-                });
-
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java]: Forgot this one, need to handle if the deletion lag and trigger size are absent at namespace level, the broker level configuration should be used.

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPolicies.java [hunk 4]
```diff
@@ -39,11 +39,15 @@
     public final static int DEFAULT_OFFLOAD_MAX_THREADS = 2;
     public final static String[] DRIVER_NAMES = {"S3", "aws-s3", "google-cloud-storage", "filesystem"};
     public final static String DEFAULT_OFFLOADER_DIRECTORY = "./offloaders";
+    public final static long DEFAULT_OFFLOAD_OFFLOAD_THRESHOLD = -1;
+    public final static Long DEFAULT_OFFLOAD_OFFLOAD_DELETION_LAG_MS = null;
```
Comment (gaoran10) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPolicies.java]: Maybe the default value of this config could keep the original value (4 hours).

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java [hunk 5]
```diff
@@ -1453,8 +1465,26 @@ void run() throws PulsarAdminException {
...
+
+            long offloadAfterThresholdInBytes = OffloadPolicies.DEFAULT_READ_BUFFER_SIZE_IN_BYTES;
+            if (StringUtils.isNotEmpty(offloadAfterThresholdStr) ) {
+                long offloadAfterThreshold = validateSizeString(offloadAfterThresholdStr);
+                if (positiveCheck("OffloadAfterThreshold", offloadAfterThreshold)
+                        && maxValueCheck("OffloadAfterThreshold", offloadAfterThreshold, Long.MAX_VALUE)) {
+                    offloadAfterThresholdInBytes = new Long(offloadAfterThreshold);
+                }
+            }
+
```
Comment (gaoran10) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java]: I have a question, there are tow CLI-commands can change the managedLedgerOffloadTreshold config and the managedLedgerOffloadDeletionLagMs config, if we use the command ```set-offload-threshold``` or ```set-offload-deletion-lag``` to set the config first, then we use the command ```set-offload-policies``` to set the bucket except for managedLedgerOffloadTreshold and managedLedgerOffloadDeletionLagMs, they will be overridden to default.

**Comment by gaoran10** [comment]
Looks good to me! 

Could you fix the description of the command ```set-offload-policies``` and ```get-offload-policies``` int the doc ```reference-pulsar-admin.md```.

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPolicies.java (hunk 4 above)
Comment (KannarFr) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/OffloadPolicies.java]: As it's deletion configuration I prefer set the default value to no deletion. WDYT?

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java (hunk 5 above)
Comment (KannarFr) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java]: "they will be overridden to default.", can you link where override is done?

**Comment by KannarFr** [comment]
I am not sure the failing tests are linked to this PR? Isn't it?

**Comment by KannarFr** [comment]
After discuss with @sijie, I prefer to create a fresh PR on https://github.com/apache/pulsar/pull/6422.

**Pull request closed without merging**